src/transfer.jl

```diff
@@ -120,36 +146,29 @@ end
 Return the Fourier coefficients for `ψk · e^{i q·r}` in the basis of `kpt_out`, where `ψk`
 is defined on a basis `kpt_in`.
 """
-function multiply_by_expiqr(basis, kpt_in, q, ψk; transfer_fn=transfer_blochwave_kpt)
-    kcoords = getfield.(basis.kpoints, :coordinate)
+function multiply_by_expiqr(basis, kpt_in, q, ψk)
+    indices_kpints_σ = findall(e -> e.:spin == kpt_in.spin, basis.kpoints)
+    kcoords = getfield.(basis.kpoints, :coordinate)[indices_kpints_σ]
     shifted_kcoord = kpt_in.coordinate .+ q  # coordinate of ``k``-point in ℝ
-    (index, ΔG) = find_equivalent_kpt(kcoords, shifted_kcoord)
-    kpt_out = basis.kpoints[index]
-    return transfer_fn(ψk, basis, kpt_in, kpt_out, ΔG)
+    index, ΔG = find_equivalent_kpt(kcoords, shifted_kcoord)
+    kpt_out = basis.kpoints[indices_kpints_σ[index]]
+    return transfer_blochwave_kpt(ψk, basis, kpt_in, kpt_out, ΔG)
 end
 
 """
-Transfer Bloch wave between two basis sets. Limited feature set.
+Return an `ordering` function that reorders the `kpoints`. That is for each `kpoint` of the
+`basis`: `kpoints[ik].coordinate + q = ordering(kpoints)[ik].coordinate`.
```

Comment:
is it really necessary to cache that information (rather than recompute it on the fly)?

Reply:
that logic is a bit convoluted (it's basically a two-stage function)

Reply:
What I meant was that basically it looks like f(x) = x -> x+y, which then you call as fx = f(x); fx(y) but you could just define f(x,y) = x+y and call it like that?

Reply:
Come on, there aren't that many kpoints, it shouldn't be that costly, certainly much less than solving the response equations. Most likely your find_equivalent_kpt function can be sped up significantly.

Reply:
But on that note you can always define a lambda ;)